pkg/volume/plugins.go

```diff
@@ -131,26 +134,53 @@ type VolumePluginMgr struct {
 
 // Spec is an internal representation of a volume.  All API volume types translate to Spec.
 type Spec struct {
-	Name                   string
-	VolumeSource           api.VolumeSource
-	PersistentVolumeSource api.PersistentVolumeSource
-	ReadOnly               bool
+	Name             string
+	Volume           *api.Volume
+	PersistentVolume *api.PersistentVolume
+	ReadOnly         bool
+}
```

Comment:
This change was necessary so that pv.Spec.Capacity was available to the bits calculating the timeout value.

Making this change as a separate PR would reduce the review required for this PR.